docs/proposals/hypernetes1.6-auth.md

```diff
@@ -0,0 +1,136 @@
+5. if check success return true and execute operation of the request. if check fail return false ,request fail.
+6. return request result 
+
+## Add user
+- Use Third Party Resources to create user resource:
+```
+apiVersion: extensions/v1beta1
+kind: ThirdPartyResource
+metadata:
+  name: user.stable.example.com
+description: "A specification of a User"
+versions:
+- name: v1
+```
+
+- Add a user object. May be there more detail user information will be set using custom fields like `username`. At least username are required.
+```
+apiVersion: "stable.example.com/v1"
+kind: User
+metadata:
+  name: alice
+username: "alice chen"
+```
+
+
+
+## auth-controller 
+
+auth-controller is responsible of updating RBAC roles and binding when
+
+- new user is created: allow the user to create/get namespace
+- new namespace is created/deleted
+- network is created/deleted
+- user is removed: delete all user related data
+
+auth -controller will watch resources including `User`, `NameSpace`, `Network` and update RBAC roles and binding using superuser.
+First  create `ClusterRole` for regular user to get permissions of `NameSpace` :
+```
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1alpha1
+metadata:
+  # "namespace" omitted since ClusterRoles are not namespaced.
+  name: namespace-creater
+rules:
+  - apiGroups: [""]
+    resources: ["namespace"]
+    verbs: ["get", "create"] # I think verbs should include "delete"，user have permission of deleting their namespace 
+```
+
+### new user are created:
+create `ClusterRoleBinding` for new regular user reference `namespace-creater` role:
+
+```
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1alpha1
+metadata:
+  name: username-namespace-creater
+subjects:
+  - kind: User 
+    name: username
+roleRef:
+  kind: ClusterRole
+  name: namespace-creater
+  apiGroup: rbac.authorization.k8s.io
+```
+### a new namespace are created:
+create `Role` namespaced:
+```
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1alpha1
+metadata:
+  namespace: namespace
+  name: access-resources-within-namespace
+rules:
+  - apiGroups: [""]
+    resources: [ResourceAll]
+    verbs: [VerbAll]
+```
+create `Rolebinding` reference to `role` namespace:
+```
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1alpha1
+metadata:
+  name: namespace-binding
+  namespace: namespace 
+subjects:
+  - kind: User
+    name: username
+roleRef:
+  kind: Role
+  name: access-resources-within-namespace
+  apiGroup: rbac.authorization.k8s.io
+```
+role and rolebinding will be removed when namespace are deleted
+### a network is created/deleted
+Which kind of resource `Network` will be defined, namespaced or non-namespaced?  If it is non-namespace, we can set permission of `Network` in `ClusterRole` when user are created. If it is namespaced resource, the permission of `Network` will be set in `Role` when namespace are created.
```

Comment:
According to our roadmap, Network is not namespace scoped. It could be associated with multiple namespaces. All pods belonging to same namespace share same network, but different namespaces may be associated with different networks. cc @feiskyer